### PR TITLE
Change definition of calculateTOF method

### PR DESCRIPTION
**Pull request by wkrzemien**
By adding missing 1/2 factor. Also, refactoring of
calculateAnihilationPoints.
Some tests are commented, and must be checked and extended.

### CODE REVIEW

**Comment by kdulski** [comment]
From what i understand from the code, my comments are like that:

 - Shift should be divided by 2, because:
when we define dR as the shift of annihilation point from the LOR middle, then TOF is proportional to 2*dR, because second hit need to travel twice the length of the shift from the middle.

- Calculation of absolute value from the versor coordinates can change LOR compeletely:
For example versor is (-1, 2, -2), then absolute value from each coordinate will change orientation of versor and further the LOR:
(-1,2, -2) -> (1, 2, 2) 
what can generate errors.

 - Calculating versor like that:
(Hit2.Pos - Hi1.Pos).Unit
already introduces the direction from the middleOfLOR, so the convention of the TOF messes everything.
Maybe this change will help
commited line 219:
TVector3 anihillationPoint(middleOfLOR.X() - std::abs(shift) * versorOnLOR.X(),
                             middleOfLOR.Y() - std::abs(shift) * versorOnLOR.Y(),
                             middleOfLOR.Z() - std::abs(shift * versorOnLOR.Z());
"-" before shift because versorOnLOR in this convention will have opposite turn to the vector from the middleOfLOR and Annihilation Point.

Tell me what You are thinking about it, and if my comments are consistent with the vision of this function.

**Comment by wkrzemien** [comment]
Thank you for your comments. 
Actually, I would like to understand one think before proceed furthere. Was the previous version correct or not?
Kamil could you send me the version you are using?

**Comment by wkrzemien** [comment]
@kkacprzak - the old version was introduced long time ago as probably temporary but it was adapted...
The new version I wrote in ~30 minutes, so it can be wrong :-).
The main issue for me was that everything was calculated from the middle of LOR, yet the shift was calculated without 1/2 factor. We spent some time  checking some simple examples and came to the conclusion that something is wrong.

**Comment by kdulski** [comment]
@wkrzemien I will send You my version of reconstruction by mail. Additionally it can be also seen at
https://github.com/kdulski/JPetFramework/blob/master/EventCategorizerTools.cpp
on the end of the file. It is very old version but works for me.

New commit seems ok, but I have one question and one comment to line 248 of the new commit in TOF calculation:
 - Is times negative? If so, the formula is correct, but maybe it will be helpful for future to comment one line, that the tof is (time1 - time2)/2   [when we assume that time1 < time2], because we have negative times, and only in that form resulting TOF value is non-negative.
 - Maybe, to avoid troubles with times, it would be helpful to rewrite line 248 as std::abs( (time1 - time2)/2 ) ? [If TOF convention is abandoned].

**Comment by wkrzemien** [comment]
@kkacprzak can you have a look again at it now?
@wictus  the same

**Comment by wkrzemien** [comment]
@kkacprzak  Could you check the code?

File: LargeBarrelAnalysis/EventCategorizerTools.cpp [hunk 1]
```diff
@@ -211,46 +218,43 @@ double EventCategorizerTools::calculateScatteringAngle(const JPetHit& hit1, cons
...
+                             middleOfLOR.Y() + shift * versorOnLOR.Y(),
+                             middleOfLOR.Z() + shift * versorOnLOR.Z());
   return annihilationPoint;
 }
 
-/**
-* Calculation Time of flight
-*/
-double EventCategorizerTools::calculateTOF(const JPetHit& firstHit, const JPetHit& latterHit)
+double EventCategorizerTools::calculateTOFArrangedByScintilatorNr(const JPetHit& hitA, const JPetHit& hitB)
```
[the remark this answers is not in the document]
Comment (wkrzemien) [reply LargeBarrelAnalysis/EventCategorizerTools.cpp]: Well, what I have been told after some discussions that there is no convention for TOF (although I was pretty sure that there is one). I will change the name and add some short description on wiki  + link to it.

File: LargeBarrelAnalysis/EventCategorizerTools.h [hunk 2]
```diff
@@ -27,40 +27,34 @@ static const double kUndefinedValue = 999.0;
...
-  );
-  static bool stream2Gamma(
-    const JPetEvent& event, JPetStatistics& stats, bool saveHistos,
-    double b2bSlotThetaDiff, double b2bTimeDiff
-  );
-  static bool stream3Gamma(
-    const JPetEvent& event, JPetStatistics& stats, bool saveHistos,
-    double d3SlotThetaMin, double d3TimeDiff, double d3DistanceFromCenter
-  );
+  /// Tof calculated with the ordered hits with respect to scintillator number
```
[the remark this answers is not in the document]
Comment (wkrzemien) [reply LargeBarrelAnalysis/EventCategorizerTools.h]: Generally I have problem with it. I know that you do it that way, but I am not sure it is the "correct" approach. My reasoning is the following: .h is defines class interface or API, so when I look into a .h file I would like to know what are the methods available, and what they do, without  switching to .cpp. Of course some special comments concering tricky parts of code should be placed in cpp independently.. I know that we are also using Doxygen, but for this it does not matter if it is in .h or cpp.

File: LargeBarrelAnalysis/EventCategorizerToolsTest.cpp [hunk 3]
```diff
@@ -371,41 +437,179 @@ BOOST_AUTO_TEST_CASE(pointAtCenter)
...
+  JPetBarrelSlot secondSlot(2, true, "second", 90, 2);
+  secondHit.setBarrelSlot(secondSlot);
+
+  TVector3 point = EventCategorizerTools::calculateAnnihilationPoint(firstHit, secondHit);
+  BOOST_REQUIRE_CLOSE(point.X(), 0.0, 1);
+  BOOST_REQUIRE_CLOSE(point.Y(), -5.0, 1);
+  BOOST_REQUIRE_CLOSE(point.Z(), 0.0, 1);
+}
+
+TVector3 kamilAnihilationPoint( const JPetHit& Hit1, const JPetHit& Hit2)
```
[the remark this answers is not in the document]
Comment (wkrzemien) [reply LargeBarrelAnalysis/EventCategorizerToolsTest.cpp]: :-)

File: LargeBarrelAnalysis/EventCategorizerTools.h [hunk 4]
```diff
@@ -27,40 +27,34 @@ static const double kUndefinedValue = 999.0;
...
+  static double calculateTOFArrangedByScintilatorNr(const JPetHit& hitA, const JPetHit& hitB);
+  /// Tof is calculated as  time1 -time2. The order is random
```
[the remark this answers is not in the document]
Comment (wkrzemien) [reply LargeBarrelAnalysis/EventCategorizerTools.h]: See the comment above

File: LargeBarrelAnalysis/EventCategorizerTools.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (wkrzemien) [reply LargeBarrelAnalysis/EventCategorizerTools.cpp]: done

File: LargeBarrelAnalysis/EventCategorizerToolsTest.cpp [hunk 5]
```diff
@@ -371,41 +437,179 @@ BOOST_AUTO_TEST_CASE(pointAtCenter)
...
+
+/// Let's assume  circle with R= 100 cm
+/// anihilation points x= 25, y=25
+/// angle = 45
+/// hit1: x = R *cos(angle)  y = R *sin(angle)
+/// hit2: x = R *-cos(angle) y = R * -sin(angle)=
+/// cos(angle) = sin(angle) = 100 *  0.70710678118 ~= 70.710678
+/// sqrt(25*25*2) ~= 35.3553
+/// t1 = (R - sqrt(25*25 *2)) / c
+/// t2 = (R + sqrt(25*25 *2)) / c
```
[the remark this answers is not in the document]
Comment (wkrzemien) [reply LargeBarrelAnalysis/EventCategorizerToolsTest.cpp]: Here, I don't know what do you mean

**Comment by wkrzemien** [comment]
Generally, I don't see the function/method overloading as a bad thing per se, especially if it gives consistent results. In this particular case, indeed it makes it easier to test, and I would like to leave it.

**Comment by kkacprzak** [comment]
Ok, good to go.

**Pull request merged**